src/scripts/migrateToPostgres.ts

```diff
@@ -67,62 +73,131 @@ async function main() {
+        await targetDs.upsertEvent(event);
         log.info(`Migrated event ${event.eventId} ${event.slackTs} (${i + 1}/${allEvents.length})`);
-    });
-
-    const teamMigrations = allTeams.map(async (team, i) => {
-        await pgres.upsertTeam(team);
-        log.info(`Migrated team ${team.id} ${team.name} (${i + 1}/${allTeams.length})`);
-    });
+    }));
 
-    const slackUserMigrations = allSlackUsers.map(async (user, i) => {
+    const slackUserMigrations = () => Promise.all(allSlackUsers.map(async (user, i) => {
         // tslint:disable-next-line: no-any
-        const ghost = SlackGhost.fromEntry(null as any, user, null);
-        await pgres.upsertUser(ghost);
+        let ghost = SlackGhost.fromEntry(null as any, user, null);
+        if (!ghost.slackId || !ghost.teamId) {
+            const localpart = ghost.userId.split(":")[0];
+            // TODO: we are making an assumption here that the prefix ends with _
+            const parts = localpart.substr(USER_PREFIX.length + 1).split("_"); // Remove any prefix.
+            // If we encounter more parts than expected, the domain may be underscored
+            while (parts.length > 2) {
+                parts[0] = `${parts.shift()}_${parts[0]}`;
+            }
+            const existingTeam = readyTeams.find((t) => t.domain === parts[0]);
+            if (!existingTeam) {
+                log.warn("No existing team could be found for", ghost.userId);
+                return;
+            }
+            user.slack_id = parts[1];
+            user.team_id = existingTeam!.id;
+            // tslint:disable-next-line: no-any
+            ghost = SlackGhost.fromEntry(null as any, user, null);
+        }
+        await targetDs.upsertUser(ghost);
         log.info(`Migrated slack user ${user.id} (${i + 1}/${allSlackUsers.length})`);
-    });
+    }));
 
-    const matrixUserMigrations = allMatrixUsers.map(async (user, i) => {
+    const matrixUserMigrations = () => Promise.all(allMatrixUsers.map(async (user, i) => {
         const mxUser = new MatrixUser(user.id, user);
         // tslint:disable-next-line: no-any
-        await pgres.storeMatrixUser(mxUser);
+        await targetDs.storeMatrixUser(mxUser);
         log.info(`Migrated matrix user ${mxUser.getId()} (${i + 1}/${allMatrixUsers.length})`);
-    });
-
-    try {
-        await Promise.all(
-            roomMigrations.concat(
-                eventMigrations,
-                teamMigrations,
-                slackUserMigrations,
-                matrixUserMigrations,
-            ),
-        );
-        log.info("Completed migration");
-    } catch (ex) {
-        log.error("An error occured while migrating databases:");
-        log.error(ex);
-        log.error("Your existing databases have not been modified, but you may need to drop the postgres table and start over");
-    }
+    }));
+    await eventMigrations();
+    log.info("Finished eventMigrations");
```

Comment:
Would be good to pair these with "Starting eventMigrations" and such, to show progress.